### ISSUE TITLE
Blank config screen for Nextcloud Office, and no connection made to CODE running separately

### ISSUE THREAD

**Issue by jogojapan** [nextcloud/richdocuments#4363]
**Describe the bug**
Nextcloud Office ("richdocuments") shows nothing in its admin panel. When running `occ richdocuments:activate-config` on the commandline, it detects the Collabora server, but it never connects to it. There is one potentially suspicious log message in the Nextcloud log, "Invalid defaultDaemonConfig data provided" (see details below).

**To Reproduce**
1. Docker-compose install of nextcloud, see docker-compose.yml and stack.env below
2. Separate docker-compose install of an Nginx reverse proxy, MariaDB, Redis. I can provide detailed config.
3. Separate docker-compose install of Collabora, see yml below
4. Confirmed that I can access nextcloud and collabora through the reverse proxy, on docs.mydomain.com and collabora.mydomain.com, respectively. Nextcloud 30 comes up and I can set up admin password etc. `https://collabora.mydomain.com/browser/dist/admin/adminLog.html` also allows me to log in, and I can see the log (which is empty and remains empty).
5. Confirmed that I can access Collabora from within the Nextcloud container:
```bash
$ docker exec --user www-data nextcloud-app-1 php -r '$fp = fsockopen("collabora.mydomain.com", 443); if ($fp) { echo "Connected\n"; fclose($fp); } else { echo "Connection failed\n"; }'

Connected
```
6. Confirmed that I can access Nextcloud from within the Collabora container:
```bash
$ docker exec shared-collabora curl -s https://docs.mydomain.com/apps/dashboard/

{"message":"Current user is not logged in"}
```
7. Log into Nextcloud as admin, Go to Apps and "download and enable" Nextcloud Office
8. Go to Admin Settings => Office. It shows a blank screen with no information
9. Attempt to activate it through the command line:
```bash
docker exec --user www-data nextcloud-app-1 php occ richdocuments:activate-config

✓ Reset callback url autodetect
Checking configuration
🛈 Configured WOPI URL: https://collabora.mydomain.com
🛈 Configured public WOPI URL: https://collabora.mydomain.com
🛈 Configured callback URL: 

✓ Fetched /hosting/discovery endpoint
✓ Valid mimetype response
✓ Valid capabilities entry
✓ Fetched /hosting/capabilities endpoint
✓ Detected WOPI server: Collabora Online Development Edition 24.04.10.2

Collabora URL (used for Nextcloud to contact the Collabora server):
  https://collabora.mydomain.com
Collabora public URL (used in the browser to open Collabora):
  https://collabora.mydomain.com
Callback URL (used by Collabora to connect back to Nextcloud):
  autodetected (will use the same URL as your user for browsing Nextcloud)
```

10. Check the Admin => Office page again, still a blank screen
11. Clicking on an .ods document in the Files section results in the file being downloaded and nothing logged in Collabora. Right-click and "edit online" results in an error shown in the browser: "There is no plugin available to display this file type"

***Docker-compose for nextcloud:***
```yaml
services:
  app:
    image: nextcloud:fpm-alpine
    restart: always
    ports:
      - 7077:9000
    volumes:
      - /volume1/nextcloud/shared_with_nginx:/var/www/html
      - /volume1/nextcloud/custom_apps:/var/www/html/custom_apps
      - /volume1/nextcloud/config:/var/www/html/config
      - /volume1/nextcloud/data:/var/www/html/data
      # theme - /volume1/nextcloud/mytheme:/var/www/html/themes/mytheme
    networks:
      - shared-backend
    env_file:
      - stack.env

  cron:
    image: nextcloud:fpm-alpine
    restart: always
    volumes_from:
      - app
    networks:
      - shared-backend
    entrypoint: /cron.sh

networks:
  shared-backend:
    external: true
```

***stack.env for nextcloud:***
```
MYSQL_DATABASE=nextcloud
MYSQL_USER=nextcloud
MYSQL_PASSWORD=mypassword
MYSQL_HOST=shared-mariadb
NEXTCLOUD_ADMIN_USER=myroot
NEXTCLOUD_ADMIN_PASSWORD=myrootpassword
REDIS_HOST=shared-redis
SMTP_HOST=email-smtp.eu-west-1.amazonaws.com
SMTP_SECURE=tls
SMTP_PORT=587
SMTP_AUTHTYPE=LOGIN
SMTP_NAME=myawskey
SMTP_PASSWORD=myawssecret
MAIL_FROM_ADDRESS=nextcloud@mydomain.com
TRUSTED_PROXIES=192.168.2.224
```

***docker-compose.yml for collabora:***
```yaml
  collabora:
    image: collabora/code:latest
    container_name: shared-collabora
    restart: always
    ports:
      - 9980:9980
    environment:
      - domain=docs.mydomain.com
      - username=myadmin
      - password=myadminpw
      - extra_params=--o:ssl.enable=false --o:ssl.termination=true
    cap_add:
      - MKNOD
    networks:
      - shared-backend

networks:
  shared-backend:
    external: true
```

**Expected behavior**
I expected:
1. The Admin => Office screen to show some information. Some of the documentation shows screenshots with a green light for a successful connection, so I expected something like that.
2. Documents to be loaded into Collabora for collaborative editing

**Screenshots**
N/A

**Client details:**
 - OS: Fedora Linux 41, Kernel 6.12.5
 - Browser: Firefox 133.03

## Server details

**Operating system**:
Asustor NAS, running Linux asustor 5.13.x SMP

**Web server:**
Nginx 1.27.3

**Database:**
MariaDB 11.4.4

**PHP version:**

**Nextcloud version:**
Nextcloud Hub 9 30.0.4

**Version of the richdocuments app**
8.5.3

**Version of Collabora Online**
COOLWSD
24.04.10.2 (git hash: [a4b67a7664](https://github.com/CollaboraOnline/online/commits/a4b67a7664))
Collabora Office 24.04.10.2 (git hash: [9d0c2a6](https://hub.libreoffice.org/git-core/9d0c2a6))

**Configuration of the richdocuments app**
```
{
    "apps": {
        "richdocuments": {
            "enabled": "yes",
            "installed_version": "8.5.3",
            "public_wopi_url": "https:\/\/collabora.mydomain.com",
            "types": "prevent_group_restriction",
            "wopi_callback_url": "",
            "wopi_url": "https:\/\/collabora.mydomain.com"
        }
    }
}
```

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
When (as admin user) clicking on "Apps", it logs the one line below:
```
{"reqId":"ERqZL4i8m6lIrO8wyCWU","level":2,"time":"2024-12-29T15:27:58+00:00","remoteAddr":"80.144.5.201","user":"myroot","app":"no app in context","method":"GET","url":"/settings/apps","message":"Invalid defaultDaemonConfig data provided to provideInitialState by settings","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:133.0) Gecko/20100101 Firefox/133.0","version":"30.0.4.1","data":[]}
```

When going into settings and viewing the Nextcloud Office settings (which only return a blank screen), nothing is logged.

The docker logs shows various PUT, GET and POST requests:
```
172.26.0.1 -  29/Dec/2024:15:26:01 +0000 "PUT /ocs/v2.php" 200
172.26.0.1 -  29/Dec/2024:15:26:01 +0000 "GET /ocs/v2.php" 304
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "REPORT /remote.php" 207
172.26.0.1 -  29/Dec/2024:15:26:03 +0000 "GET /ocs/v2.php" 200
172.26.0.1 -  29/Dec/2024:15:26:04 +0000 "GET /ocs/v2.php" 200
172.26.0.1 -  29/Dec/2024:15:26:04 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:04 +0000 "PROPFIND /remote.php" 207
172.26.0.1 -  29/Dec/2024:15:26:04 +0000 "GET /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:04 +0000 "GET /ocs/v2.php" 200
172.26.0.1 -  29/Dec/2024:15:26:04 +0000 "PUT /ocs/v2.php" 200
172.26.0.1 -  29/Dec/2024:15:26:04 +0000 "GET /ocs/v2.php" 200
172.26.0.1 -  29/Dec/2024:15:26:04 +0000 "GET /index.php" 404
172.26.0.1 -  29/Dec/2024:15:26:05 +0000 "GET /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:05 +0000 "PROPFIND /remote.php" 207
172.26.0.1 -  29/Dec/2024:15:26:05 +0000 "GET /ocs/v2.php" 200
172.26.0.1 -  29/Dec/2024:15:26:05 +0000 "GET /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:05 +0000 "PUT /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:05 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:05 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:06 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:07 +0000 "PROPFIND /remote.php" 207
172.26.0.1 -  29/Dec/2024:15:26:08 +0000 "GET /ocs/v2.php" 200
172.26.0.1 -  29/Dec/2024:15:26:08 +0000 "PROPFIND /remote.php" 207
172.26.0.1 -  29/Dec/2024:15:26:08 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:08 +0000 "PUT /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:08 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:08 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:08 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:12 +0000 "GET /remote.php" 200
172.26.0.1 -  29/Dec/2024:15:26:13 +0000 "GET /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:13 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:18 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:23 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:26 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:28 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:34 +0000 "POST /index.php" 200
172.26.0.1 -  29/Dec/2024:15:26:34 +0000 "GET /ocs/v2.php" 304
```
</details>

**Comment by jogojapan** [comment]
For completeness' sake, I am attaching a screenshot of the blank Configuration screen for Nextcloud Office.

![Image](https://github.com/user-attachments/assets/1901dd30-f427-445c-aa08-d4af013d4da5)

Looking at the HTML source of the page, I can see that there is in fact some information, in hidden divs:

```html
<div id="app-content"  data-active-section-id="richdocuments"  data-active-section-type="admin" >
	<div id="admin-vue" data-initial="{&quot;wopi_url&quot;:&quot;https:\/\/collabora.mydomain.com&quot;,&quot;public_wopi_url&quot;:&quot;https:\/\/collabora.mydomain.com&quot;,&quot;wopi_callback_url&quot;:&quot;&quot;,&quot;wopi_allowlist&quot;:&quot;&quot;,&quot;edit_groups&quot;:&quot;&quot;,&quot;use_groups&quot;:&quot;&quot;,&quot;doc_format&quot;:&quot;&quot;,&quot;external_apps&quot;:&quot;&quot;,&quot;canonical_webroot&quot;:&quot;&quot;,&quot;disable_certificate_verification&quot;:false,&quot;templates&quot;:[{&quot;id&quot;:379,&quot;name&quot;:&quot;Empty&quot;,&quot;type&quot;:&quot;document&quot;,&quot;extension&quot;:&quot;odt&quot;},{&quot;id&quot;:382,&quot;name&quot;:&quot;Empty&quot;,&quot;type&quot;:&quot;drawing&quot;,&quot;extension&quot;:&quot;odg&quot;},{&quot;id&quot;:381,&quot;name&quot;:&quot;Empty&quot;,&quot;type&quot;:&quot;presentation&quot;,&quot;extension&quot;:&quot;odp&quot;},{&quot;id&quot;:380,&quot;name&quot;:&quot;Empty&quot;,&quot;type&quot;:&quot;spreadsheet&quot;,&quot;extension&quot;:&quot;ods&quot;}],&quot;templatesAvailable&quot;:true,&quot;settings&quot;:{&quot;enabled&quot;:true,&quot;installed_version&quot;:&quot;8.5.3&quot;,&quot;public_wopi_url&quot;:&quot;https:\/\/collabora.mydomain.com&quot;,&quot;types&quot;:&quot;prevent_group_restriction&quot;,&quot;wopi_callback_url&quot;:&quot;&quot;,&quot;wopi_url&quot;:&quot;https:\/\/collabora.mydomain.com&quot;},&quot;demo_servers&quot;:null,&quot;web_server&quot;:&quot;nginx\/1.27.3&quot;,&quot;os_family&quot;:&quot;Linux&quot;,&quot;platform&quot;:&quot;x86_64&quot;,&quot;fonts&quot;:[&quot;AmaticSC-Bold.ttf&quot;,&quot;AmaticSC-Regular.ttf&quot;,&quot;Caveat-wght.ttf&quot;,&quot;Comfortaa-wght.ttf&quot;,&quot;EBGaramond-wght.ttf&quot;,&quot;Lexend-wght.ttf&quot;,&quot;Lobster-Regular.ttf&quot;,&quot;Lora-wght.ttf&quot;,&quot;Montserrat-wght.ttf&quot;,&quot;Nunito-wght.ttf&quot;,&quot;Oswald-wght.ttf&quot;,&quot;Pacifico-Regular.ttf&quot;,&quot;PlayfairDisplay-wght.ttf&quot;,&quot;Roboto-wdtg-wght.ttf&quot;,&quot;RobotoMono-wght.ttf&quot;,&quot;Spectral-Regular.ttf&quot;]}"></div>

<form class="section hidden" id="richdocuments-templates" method="post" action="[/template/](view-source:https://docs.mydomain.com/template/)">
	<input name="files" class="hidden-visually" id="add-template" type="file" />
	<h2>
		Global templates		<label for="add-template" class="icon-add" title="Add a new template"></label>
	</h2>
	<div id="emptycontent" class="hidden">
		<div class="icon-file"></div>
		<h2>
			No templates defined.		</h2>
		<label for="add-template">Add a new one?</label>
	</div>
	<ul class="1">
		<li class="hidden template-model">
			<figure>
				<img src="" alt="template preview" />
				<figcaption></figcaption>
			</figure>
			<a href="" class="delete-template icon-delete"></a>
			<div class="delete-cover"></div>
		</li>
					<li data-filename="Empty">
				<figure>
											<div class="templatePlaceholder"></div>
										<figcaption>Empty</figcaption>
				</figure>
								<div class="delete-cover"></div>
			</li>
					<li data-filename="Empty">
				<figure>
											<div class="templatePlaceholder"></div>
										<figcaption>Empty</figcaption>
				</figure>
								<div class="delete-cover"></div>
			</li>
					<li data-filename="Empty">
				<figure>
											<div class="templatePlaceholder"></div>
										<figcaption>Empty</figcaption>
				</figure>
								<div class="delete-cover"></div>
			</li>
					<li data-filename="Empty">
				<figure>
											<div class="templatePlaceholder"></div>
										<figcaption>Empty</figcaption>
				</figure>
								<div class="delete-cover"></div>
			</li>
			</ul>
</form>
```

**Comment by jogojapan** [comment]
I figured it out.

The problem was that the container for reverse proxy (Nginx) was configured like this (showing the `volumes` config only):

```yaml
    volumes:
      - /volume1/nginx/secrets:/etc/letsencrypt
      - /volume1/nginx/user_conf.d:/etc/nginx/user_conf.d
      - /volume1/nginx/nginx.conf:/etc/nginx/nginx.conf
      - /volume1/nextcloud/shared_with_nginx:/var/www/html
```

So it was loading the `/var/www/html` volume it needs to share with the Nextcloud container, as described in the docker-compose examples, e.g. [here](https://github.com/nextcloud/docker/blob/85eb58a82803a67c9f5567e7c5d1610e12b34ace/.examples/docker-compose/with-nginx-proxy/mariadb/fpm/compose.yaml#L40).

I add assumed that it would suffice if Nginx had access to the main `/var/www/html` directory, and not the `data`, `config` and `custom_apps` subdirectories, which I am mounting as separate volumes.

Turns out Nginx needs access to `apps` (which it gets automatically, because Nextclouds creates it in the `/var/www/html` main directory), and also to `custom_apps`.

The reason it showed a blank screen for the Nextcloud Office configuration, and a blank icon for the app, was that Nginx couldn't access any of the static files. In fact, it would have had the same problem with any further custom app I was going to install. Office just happened to be the first one.

So the **solution** is to mount `custom_apps` by adding it to the `volumes` config above:

```yaml
      - /volume1/nextcloud/custom_apps:/var/www/html/custom_apps
```

I hope it does not also need the `config` and `data` volumes. I could add those in as well, but I don't really want the reverse proxy (which is shared with many other services running on that server) to mount the data folder.
